File: 01_correction/detection.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    std::vector<std::string> a = {"01010100", "01001011", "10110111", "10101000"};
    std::vector<size_t> result;
    for (size_t i = 0; i < a.size(); i++) {
        for (size_t j = i + 1; j < a.size(); j++) {
            size_t temp = 0;
            for (size_t k = 0; k < a[i].size(); k++) {
                if (a[i][k] != a[j][k])
                    temp++;
            }
            result.push_back(temp);
        }
    }
    size_t min = a[0].size();
    for (const auto& x : result) {
        if (x < min)
            min = x;
    }

    std::cout << "d(V) = " << min << std::endl;
}

File: 01_correction/detection.py
import sys

argc = len(sys.argv) - 1
argv = sys.argv[1:]

if argc < 2:
    print("Usage:   detection.py [CODES...]")
    print("Example: detection.py 01010100 01001011 10110111 10101000")
    print("Error:   At least two codes are required.")
    exit(1)

a = argv
result = []

for i in range(len(a)):
    for j in range(i + 1, len(a)):
        temp = 0
        for k in range(len(a[i])):
            if a[i][k] != a[j][k]:
                temp += 1
        result.append(temp)

d = len(a[0])

for x in result:
    if x < d:
        d = x

print("d(V) =", d)
